merlin/models/loader/backend.py

```diff
@@ -40,12 +40,12 @@ def _augment_schema(
         properties = cs.properties
         if sparse_max and col in sparse_max:
             properties["value_count"] = {"max": sparse_max[col]}
+        properties["is_list"] = True
+        properties["is_ragged"] = not sparse_as_dense
```

Comment:
I am thinking out load. so if we set `sparse_as_dense` to False (default), does that mean we do not have a fixed length list, so `is_ragged` becomes True.  But my confusion is, if someone sets sparse_to_dense to True, that means, we have sparse input representation, and that means we have list columns where these list columns could be fixed or variable length. Am I wrong? But due to this line ` properties["is_ragged"] = not sparse_as_dense` `is_ragged` becomes False.  So my question, if sparse_to_dense  should not be set to True if the list columns are variable length (ragged)? sorry If I am misunderstanding the relation between `sparse_as_dense` arg and list columns.

Reply:
> sorry If I am misunderstanding the relation between `sparse_as_dense` arg and list columns
I'm not as familiar with the history of `sparse_as_dense` as you are, so it's just as likely to be me misunderstanding the relationship. Let's try to figure it out together though! 😄 

> if we set `sparse_as_dense` to False (default), does that mean we do not have a fixed length list, so is_ragged becomes True
If I understand what `sparse_as_dense` does (and I may not), it converts ragged tensors to fixed length tensors by padding them, yeah? If that's true, I think `sparse_as_dense` should probably be called either `ragged_as_dense` or `pad`, which would make this easier to think about. If `sparse/ragged_as_dense/pad` is `False`, that means we're not padding, which means that any ragged list remains ragged.

> if someone sets `sparse_to_dense` to True, that means, we have sparse input representation, so that we have list columns where these list columns could be fixed or variable length
If `sparse/ragged_to_dense/pad` is True, I think that means we'd pad any ragged list inputs, so ragged lists would be converted to fixed length, after which all lists would be fixed length (i.e. `is_ragged=False`.)

> due to this line `properties["is_ragged"] = not sparse_as_dense` is_ragged becomes False. So my question, should sparse_to_dense not be set to True if the list columns are variable length (ragged)?
The way I understood this was `sparse/ragged_as_dense/pad = False` implies that ragged lists are not padded and therefore stay ragged, while `sparse/ragged_as_dense/pad = True` implies that ragged lists are padded and therefore become fixed length. I'm not sure if I have that right though...

Reply:
Yes @karlhigley, the `sparse_as_dense` API of the older NVT dataloader was intended to work like that, padding list columns by converting them from Ragged to Dense when `sparse_as_dense=True` 
It makes sense to me this change in `_augment_schema()`, as it only sets `is_ragged` property to sparse/list columns, so that `is_ragged` needs to be either `False` or `True`.